cmd/server/app/serve.go

```diff
@@ -120,9 +121,13 @@ var serveCmd = &cobra.Command{
 
 		s.ConsumeEvents(aggr)
 
+		tsmdw := logger.NewTelemetryStoreWMMiddleware(l)
+
 		exec, err := engine.NewExecutor(ctx, store, &cfg.Auth, evt,
 			engine.WithProviderMetrics(providerMetrics),
-			engine.WithAggregatorMiddleware(aggr))
+			engine.WithMiddleware(aggr.AggregateMiddleware),
+			engine.WithMiddleware(tsmdw.TelemetryStoreMiddleware),
```

Comment:
Do we want to also collect telemetry from the reconciler?

Reply:
I'm not sure how much value do we get from that. But if we do, then we'd need dedicated middleware, as it does not use the entity info wrapper.